public/ServiceWorker/index.js

```diff
@@ -0,0 +1,29 @@
+self.addEventListener('install', event => {
+    event.waitUntil(
+        caches.open('cache-v1-pwa')
+            .then(cache => {
+                return cache.addAll([
+                    '/'
+                ])
+            })
+            .then(() => self.skipWaiting())
+    )
+})
+self.addEventListener('activate', event => {
+    event.waitUntil(
+        caches.keys()
+            .then(keys => {
+                return Promise.all(
+                    keys.filter(key => key !== 'cache-v1-pwa')
+                        .map(key => caches.delete(key))
+                )
+            })
+            .then(() => self.clients.claim())
+    )
+})
+self.addEventListener('fetch', event => {
+    event.respondWith(
+        caches.match(event.request)
+            .then(response => response || fetch(event.request))
+    )
+})
```

Comment:
This code sets up an event listener for the fetch event on the service worker self. When the service worker intercepts a fetch request from a client, the callback function is executed.